lib/spack/spack/schema/packages.py

```diff
@@ -41,6 +41,8 @@
                             "spec": {"type": "string"},
                             "message": {"type": "string"},
                             "when": {"type": "string"},
+                            "root_only": {"type": "boolean"},
+                            "applies_for_externals": {"type": "boolean"},
```

Comment:
I think this should be a bit more flexible / extensible and we should use a list of strings rather than boolean options, e.g.:

```yaml
packages:
  all:
    require:
    - spec: "@1.0"
      include:: ["externals", "roots"]
```

I'm not sure these are the only elements we'd want but it's a start.  The default would be to include everything.